internal/apptoken/options.go

```diff
@@ -55,3 +58,17 @@ func WithLimit(_ context.Context, limit int) Option {
 		return nil
 	}
 }
+
+// WithExpirationInterval provides an option to specify an expiration interval
+// for an AppToken
+func WithExpirationInterval(ctx context.Context, maxSeconds uint32) Option {
+	const op = "apptoken.WithExpirationInterval"
+	return func(o *options) error {
+		switch {
+		case maxSeconds == 0:
+			return errors.New(ctx, errors.InvalidParameter, op, "invalid expiration interval (equal to zero)")
+		}
```

Comment:
Curious why we return an error here even when it is a valid case considering it is an optional param. Is it so the caller doesn't unintentionally pass `0` value?

Reply:
correct.  The caller is intentionally specifying an interval and zero isn't a
valid interval.